### PR TITLE
Align isolated network capabilities with other operators

### PR DESCRIPTION
**Pull request by stuggi**
* If needed, service pods can be attach to networks using the `networkAttachments` list in the CR. Its a list of pre-created net-attach-defs names. Per default the pods won't get attached.

```
spec:
...
      networkAttachments:
      - storage
```
Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/133

* introduce externalEndpoints Parameters to configure metallb service using the pre-created metallb IPAddressPool and L2Advertisement . 

E.g. to expose the api on an internalapi network which
* should be registered as the internal endpoint
* its metallb ipaddresspool is configured in an osp-internalapi object
* and request a specific loadbalancerIP which is shared with other services (list of IPs for multi stack ipv4/ipv6)

```
...
spec:
...
      externalEndpoints:
      - endpoint: internal
        loadBalancerIPs:
        - 172.17.0.202
        ipAddressPool: osp-internalapi
        sharedIP: true
```

* removes registration of admin endpoint

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/160

### CODE REVIEW

**Review by hjensas — CHANGES_REQUESTED** [review]
Looks ok, just the nit's in the README.

File: README.md [hunk 1]
```diff
@@ -57,6 +57,69 @@ spec:
...
+metadata:
+  name: osp-internalapi
+  namespace: metallb-system
+spec:
+  addresses:
+  - 172.17.0.200-172.17.0.210
+  autoAssign: false
+```
+
+The following represents an example of Glance resource that can be used
```
[the remark this answers is not in the document]
Comment (stuggi) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -57,6 +57,69 @@ spec:
...
+to trigger the service deployment, and have the internal glanceAPI endpoint
```
[the remark this answers is not in the document]
Comment (stuggi) [reply README.md]: done

**Review by hjensas — COMMENTED** [review]
Thanks, looks good.

**Review by hjensas — APPROVED** [review]

**Pull request merged**